build/authors.py

```diff
@@ -2,9 +2,116 @@
 Process author table
 """
 
-import collections
 import pandas
+import numpy as np
 
+from collections import (
+    defaultdict,
+    OrderedDict,
+)
+
+
+def initials(full_name):
+    """
+    Return author intials in the form F.L. for First Last.  Accomodates an
+    aribtrary number of whitespace-separated strings.  Converts periods to
+    whitespace so that multiple middle initials are retained.
+    """
+    full_name = full_name.replace('.', ' ')
+    name_tokens = full_name.strip().split()
+    # Keeps only the first part of hyphenated names
+    name_tokens = [name[0] + '.' for name in name_tokens]
+    return ''.join(name_tokens)
+
+def format_list(tokens):
+    """
+    Format a list of string tokens such that a pair [X, Y] is formatted 'X and
+    Y' and a list [X, Y, Z] becomes 'X, Y, and Z'.  Single tokens are returned
+    as-is.
+    """
+    if len(tokens) <= 2:
+        return ' and '.join(tokens)
+    else:
+        tokens[-1] = 'and ' + tokens[-1]
+        return ', '.join(tokens)
+
+def get_funding(author_df):
+    """
+    Returns a dictionary mapping funders ('GBMF', 'NIH', 'NSF', 'OTHER') to
+    award and author text.  An author may have more than one funding
+    source and multiple authors may have the same funding source.
+    """
+    # Initially create a map from funders to dictionaries that map an award to
+    # the authors supported by that award
+    # funding_info data structure has the form
+    # {funder1: {award1: [author1], award2: [author2, author3]}}
+    funding_info = defaultdict(lambda: defaultdict(list))
```

Comment:
I think you're looking for the `.setdefault()` method of `dict` rather than this.